brasilio_auth/forms.py

```diff
@@ -24,7 +26,13 @@ class Meta:
 
     def clean_username(self):
         username = self.cleaned_data.get("username", "")
-        return username.lower()
+        username = username.lower()
+        non_valid_chars = re.search(r"[^a-z0-9_]", username)
```

Comment:
Ao usar expressões regulares em funções que serão executadas repetidas vezes, o ideal é compilar a expressão no início do comprar e usar a versão compilada. Nesse caso, a execução de uma regexp compilada demora apenas 24% do tempo de uma não compilada:

```
In [1]: import re

In [2]: %timeit re.search(r"[^a-z0-9_]", "turicas")
606 ns ± 30.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: REGEXP_USERNAME = re.compile(r"[^a-z0-9_]")

In [4]: %timeit REGEXP_USERNAME.search("turicas")
145 ns ± 9.22 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: 145/606
Out[5]: 0.23927392739273928
```

Reply:
solved by 985cc1d3bae5294d56f2